BotFramework/Logging/Log.cs

```diff
@@ -0,0 +1,151 @@
+         CreateLog(LogType.Error, filePath, function, message, payload);
+
+    public static void Warn(string message, string payload = "", [CallerFilePath] string filePath = "", [CallerMemberName] string function = "") =>
+         CreateLog(LogType.Warn, filePath, function, message, payload);
+
+
+    public static void Info(string message, string payload = "", [CallerFilePath] string filePath = "", [CallerMemberName] string function = "") =>
+         CreateLog(LogType.Info, filePath, function, message, payload);
+
+
+    public static void Debug(string message, string payload = "", [CallerFilePath] string filePath = "", [CallerMemberName] string function = "") =>
+         CreateLog(LogType.Debug, filePath, function, message, payload);
+
+
+    public static void Trace(string message, string payload = "", [CallerFilePath] string filePath = "", [CallerMemberName] string function = "") =>
+         CreateLog(LogType.Trace, filePath, function, message, payload);
+
+    private static string GetClassName(string filePath)
+    {
+        int index = filePath.LastIndexOf("\\") + 1;
+        return filePath.Substring(index, filePath.LastIndexOf(".") - index);
+    }
+
+    public static void CreateLog(LogType logType, string classFilepath, string function, string message, string payload = "")
+    {
+        string query = @"
+            INSERT INTO Logs (Level, Class, Function, Message, Payload, Time)
+            VALUES (@Level, @Class, @Function, @Message, @Payload, datetime());";
+
+        using (SQLiteCommand command = new SQLiteCommand(query, _connection))
+        {
+            command.Parameters.AddWithValue("@Level", (int)logType);
+            command.Parameters.AddWithValue("@Class", GetClassName(classFilepath));
+            command.Parameters.AddWithValue("@Function", function);
+            command.Parameters.AddWithValue("@Message", message);
+            command.Parameters.AddWithValue("@Payload", payload);
+
+            try
+            {
+                command.ExecuteNonQuery();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+        }
+    }
+
+    public static SQLiteDataReader ReadSQL(string query, (string paramName, object value)[]? parameters)
+    {
+        SQLiteDataReader reader = null;
+        using (SQLiteCommand command = new SQLiteCommand(query, _connection))
+        {
+            if (parameters != null)
+            {
+                foreach (var param in parameters)
+                {
+                    command.Parameters.AddWithValue(param.paramName, param.value);
+                }
+            }
+            try
+            {
+                reader = command.ExecuteReader();
+            }
+            catch (Exception e)
+            {
+                Log.Debug(e.Message);
+            }
+        }
+        return reader;
+    }
```

Comment:
```suggestion
    public static SQLiteDataReader ReadSQL(string query, params (string paramName, object value)[] parameters)
    {
        SQLiteDataReader reader = null;
        using (SQLiteCommand command = new SQLiteCommand(query, _connection))
        {
            foreach (var param in parameters)
            {
                command.Parameters.AddWithValue(param.paramName, param.value);
            }
        }
            
        try
        {
            reader = command.ExecuteReader();
        }
        catch (Exception e)
        {
            Log.Debug(e.Message);
        }
        
        return reader;
    }
```

Should work with the named `Tuple` thing going on too.